File: dev-docs/snippets/collections/list-pending-invitations.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List pending (deferred, non-expired) invitations for a collection.
export async function listPendingInvitations(
  client: JsBaoClient,
  collectionId: string,
) {
  // #region example
  const pending = await client.collections.listPendingInvitations(collectionId);
  for (const invite of pending) {
    console.log(invite.email, invite.permission, invite.expiresAt);
  }
  // #endregion example
  return pending;
}

File: dev-docs/snippets/collections/list-pending-invitations.swift
import JsBaoClient

// List pending (deferred, non-expired) invitations for a collection.
func listPendingInvitations(
  client: JsBaoClient,
  collectionId: String
) async throws {
  // #region example
  let pending = try await client.collections.listPendingInvitations(
    collectionId: collectionId
  )
  for invite in pending {
    print(invite.email, invite.permission, invite.expiresAt)
  }
  // #endregion example
  _ = pending
}
